app-ios/Sources/TimetableFeature/Content/ListView.swift

```diff
@@ -1,15 +1,118 @@
+import appioscombined
+import Assets
 import SwiftUI
+import Theme
+import Model
+import ComposableArchitecture
 
 struct TimetableListView: View {
+    struct ViewState: Equatable {
+        var timeGroupTimetableItems: [TimetableTimeGroupItems]
+
+        init(state: TimetableState) {
+            guard let timetable = state.dayToTimetable[state.selectedDay] else {
+                self.timeGroupTimetableItems = []
+                return
+            }
+            self.timeGroupTimetableItems = Set(timetable.timetableItems.map { Duration(startsAt: $0.startsAt, endsAt: $0.endsAt) })
+                .map { duration in
+                    let startsAt = duration.startsAt
+                    let endsAt = duration.endsAt
+                    let items = timetable.contents
+                        .filter { itemWithFavorite in
+                            itemWithFavorite.timetableItem.startsAt == startsAt && itemWithFavorite.timetableItem.endsAt == endsAt
+                        }
+                        .sorted {
+                            $0.timetableItem.room.sort < $1.timetableItem.room.sort
+                        }
+                    let minute = calculateMinute(
+                        startSeconds: Int(startsAt.epochSeconds),
+                        endSeconds: Int(endsAt.epochSeconds)
+                    )
+                    return TimetableTimeGroupItems(
+                        startsAt: startsAt.toDate(),
+                        endsAt: endsAt.toDate(),
+                        minute: minute,
+                        items: items
+                    )
+                }
+                .sorted {
+                    $0.startsAt == $1.startsAt ? $0.minute < $1.minute : $0.startsAt < $1.startsAt
+                }
+        }
+    }
+
+    /// `startsAt` & `endsAt` pair struct conformed to `Hashable`
+    private struct Duration: Hashable {
+        let startsAt: Kotlinx_datetimeInstant
+        let endsAt: Kotlinx_datetimeInstant
+    }
+
+    private let dateComponentsFormatter: DateComponentsFormatter = {
+        let formatter = DateComponentsFormatter()
+        return formatter
+    }()
+
+    let store: Store<TimetableState, TimetableAction>
+
     var body: some View {
-        Text("TODO: TimetableListView")
+        WithViewStore(store.scope(state: ViewState.init)) { viewStore in
+            ScrollView(.vertical) {
+                LazyVStack(spacing: 32) {
+                    ForEach(viewStore.timeGroupTimetableItems, id: \.self) { timetableTimeGroupItems in
```

Comment:
This can be omitted 👍🏼 
```suggestion
                    ForEach(viewStore.timeGroupTimetableItems) { timetableTimeGroupItems in
```

Reply:
Thanks for your info! I omitted it https://github.com/DroidKaigi/conference-app-2022/pull/402/commits/fa95b5c7a441f18c43d5ba275cafd8d2ab2e330d